db/version_set.cc

```diff
@@ -6186,7 +6191,7 @@ Status ReactiveVersionSet::ApplyOneVersionEditToBuilder(
       auto version = new Version(cfd, this, file_options_,
                                  *cfd->GetLatestMutableCFOptions(),
                                  current_version_number_++);
-      builder->SaveTo(version->storage_info());
+      s = builder->SaveTo(version->storage_info());
```

Comment:
Probably should not/ need not call `PrepareApply` and `AppendVersion` if `!s.ok()`.

Reply:
Move to #6801